File: src/main/java/kr/ac/kumoh/k20230626/javawebspring/repository/InMemoryLottoRepository.java
package kr.ac.kumoh.k20230626.javawebspring.repository;

import kr.ac.kumoh.k20230626.javawebspring.model.LottoNumber;
import org.springframework.stereotype.Repository;

@Repository
public class InMemoryLottoRepository implements LottoRepository{
    private LottoNumber storedLottoNumber;
    @Override
    public LottoNumber save(LottoNumber lottoNumber) {
        this.storedLottoNumber = lottoNumber;
        return storedLottoNumber;
    }

    @Override
    public LottoNumber find() {
        return this.storedLottoNumber;
    }
}

File: src/main/kotlin/kr/ac/kumoh/k20230626/lotto/repository/InMemoryLottoRepository.kt
package kr.ac.kumoh.k20230626.lotto.repository

import kr.ac.kumoh.k20230626.lotto.model.LottoNumber
import org.springframework.stereotype.Repository

@Repository
class InMemoryLottoRepository: LottoRepository {

    private var storedLottoNumber: LottoNumber? = null
    override fun save(lottoNumber: LottoNumber): LottoNumber {
        storedLottoNumber = lottoNumber
        return storedLottoNumber!!
    }


    override fun find(): LottoNumber? {
        TODO("Not yet implemented")
    }
}